src/flb_upstream.c

```diff
@@ -843,9 +844,12 @@ int flb_upstream_conn_timeouts(struct mk_list *list)
 
             if (drop == FLB_TRUE) {
                 if (u_conn->event.status != MK_EVENT_NONE) {
-                    mk_event_inject(u_conn->evl, &u_conn->event,
-                                    MK_EVENT_READ | MK_EVENT_WRITE,
-                                    FLB_TRUE);
+                    result = mk_event_inject(u_conn->evl, &u_conn->event,
+                                             MK_EVENT_READ | MK_EVENT_WRITE,
+                                             FLB_TRUE);
+                    if (result == 1) {
+                        continue;
```

Comment:
Is the side effect of not marking the connection as timed out something calculated?